deployment/cloud-run/test_routing_fixed.py

```
@@ -14,44 +15,72 @@
+    root_pattern = re.compile(r'^/?/?$')  # Matches '/', '//', or '' (empty string) - more flexible
+    root_routes = [rule for rule in app.url_map.iter_rules() if root_pattern.match(rule.rule)]
     if root_routes:
-        print("✅ Root endpoint (/) exists")
+        print("Root endpoint (/) exists")
         for route in root_routes:
             print(f"   - {route.endpoint} (methods: {route.methods})")
+            # Add assertion for pattern match
+            assert root_pattern.match(route.rule), f"Route {route.rule} does not match root pattern"
     else:
-        print("❌ Root endpoint (/) missing")
+        print("Root endpoint (/) missing")
```

Ruff (0.12.2)

`print` found

Remove `print`

(T201)